rewrite-java/src/main/java/org/openrewrite/java/format/AutoFormatVisitor.java

```diff
@@ -121,25 +131,52 @@ public J visit(@Nullable Tree tree, P p) {
             t = (JavaSourceFile) new NormalizeTabsOrSpacesVisitor<>(tabsAndIndentsStyle, stopAfter)
                     .visitNonNull(t, p);
 
-            return new TabsAndIndentsVisitor<>(tabsAndIndentsStyle, spacesStyle, stopAfter)
+            t = (JavaSourceFile) new TabsAndIndentsVisitor<>(tabsAndIndentsStyle, spacesStyle, stopAfter)
                     .visitNonNull(t, p);
+
+            return addStyleMarker(t);
         }
         return (J) tree;
     }
 
+    private JavaSourceFile addStyleMarker(JavaSourceFile t) {
+        if (!styles.isEmpty()) {
+            Set<NamedStyles> newNamedStyles = new HashSet<>(styles);
+            boolean styleAlreadyPresent = false;
+            for (NamedStyles namedStyle : t.getMarkers().findAll(NamedStyles.class)) {
+                styleAlreadyPresent = !newNamedStyles.add(namedStyle) || styleAlreadyPresent;
+            }
+            // As the order or NamedStyles matters, we cannot simply use addIfAbsent.
+            if (!styleAlreadyPresent) {
+                Markers markers = t.getMarkers().removeByType(NamedStyles.class);
+                for (NamedStyles namedStyle : newNamedStyles) {
+                    markers = markers.add(namedStyle);
+                }
+
+                return t.withMarkers(markers);
+            }
+        }
+        return t;
+    }
```

Comment:
Nothing about this is really specific to `JavaSourceFile` or the java-specific version of `AutoFormat`. This could be a method in `StyleHelper` which accepts a `Markers` or a `SourceFile`. That would then be available to all other implementations of `AutoFormat` which may want to adopt similar functionality for similar reasons.